### PR TITLE
docs: general cleanup and maintenance

### LINKED ISSUE #50462: Content of 'Work on Codebase' section before 'Open a Pull Request' section under Code Contribution in docs.
### Is your feature request related to a problem? Please describe.

According to the current order of docs under the Code contribution section, you encounter the 'Open a Pull Request' section before the 'Work on Codebase' section which has the steps to create a branch as well as the conventions for a commit. According to me, the later section should be displayed first as that is how the correct procedure goes, i.e., Branch -> Commit -> Pull Request.

### Describe the solution you'd like

A simple solution is just swapping the order of both the sections which would ease the flow for the contributor. The 'Working on Codebase' could be given a different name (Couldn't think of a appropriate name here). Also, we currently have a 'Proposing a Pull Request' sub-section in the 'Working on Codebase' section. This could simply be removed as the Pull Request section will naturally precede it in the docs after this change.

### Describe alternatives you've considered

Don't think this applies here.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

I went down a bit of a rabbit hole here...

- Updates the order of the navbar for the pull request instructions, adds a link to those instructions to a couple of "how to contribute" docs that were missing it. Closes #50462 
- Normalises the heading levels, and removes a couple of places where we were abusing headings for styling (which is an accessibility no-no)
- Adds a link in the navbar for the Docker on Windows instructions, which was apparently an orphaned page
- Moves the reply templates to a separate page, allowing the sub-headings to appear in the navbar (should make navigating to a specific template much easier)

### CODE REVIEW

**Review by Sboonny — APPROVED** [review]
LGTM

**Review by Sboonny — APPROVED** [review]
LGTM

File: docs/how-to-work-on-the-component-library.md [hunk 1]
```diff
@@ -92,6 +92,10 @@ To run tests against the component library, run the following command from the r
 pnpm run test-ui-components
 ```
 
+## Proposing a Pull Request (PR)
+
+After you've committed your changes, check here for [how to open a Pull Request](how-to-open-a-pull-request.md).
```
Comment (Sembauke) [inline docs/how-to-work-on-the-component-library.md:97]:
```suggestion
After you've committed your changes, check here [how to open a Pull Request](how-to-open-a-pull-request.md).
```

File: docs/how-to-work-on-coding-challenges.md [hunk 2]
```diff
@@ -484,14 +484,16 @@ You are also able to test one challenge individually by performing the following
...
-
 > [!TIP]
 > You can set the environment variable `LOCALE` in the `.env` to the language of the challenge(s) you need to test.
 >
 > The currently accepted values are `english` and `chinese`, with `english` being set by default.
 
-### Useful Links
+## Proposing a Pull Request (PR)
+
+After you've committed your changes, check here for [how to open a Pull Request](how-to-open-a-pull-request.md).
```
Comment (Sembauke) [inline docs/how-to-work-on-coding-challenges.md:494]:
```suggestion
After you've committed your changes, check here [how to open a Pull Request](how-to-open-a-pull-request.md).
```

File: docs/devops.md [hunk 3]
```diff
@@ -209,19 +209,19 @@ You can identify the exact version deployed by visiting the build and deployment
...
 
   **Users on the beta version will have a separate account from the production.** The beta version uses a physically separate database from production. This gives us the ability to prevent any accidental loss of data or modifications. The dev-team may purge the database on this beta version as needed.
 
-- #### There are no guarantees on the uptime and reliability of the beta platforms
+- **There are no guarantees on the uptime and reliability of the beta platforms**
 
   Deployment is expected to be frequent and in rapid iterations, sometimes multiple times a day. As a result, there will be unexpected downtime at times or broken functionality on the beta version.
 
-- #### Do not send regular users to this site as a measure of confirming a fix
+- **Do not send regular users to this site as a measure of confirming a fix**
```
Comment (Sembauke) [inline docs/devops.md:220]:
```suggestion
- **To ensure the effectiveness of the fix, it is advised not to direct regular users to this site for verification purposes.**
```

File: docs/moderator-handbook.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ You can help us organize and triage the issue reports by applying labels from [t
 
 Please pay special attention to the labels `"help wanted"` and `"first timers only"`. These are to be added to threads that you think can be opened up to potential contributors for making a pull request.
 
-A `"first timers only"` label should be applied to a trivial issue (ie a typo fix) and should include additional information. You can use this [reply template](moderator-handbook.md#first-timer-only-issues) for triage.
+A `"first timers only"` label should be applied to a trivial issue (ie a typo fix) and should include additional information. You can use this [reply template](reply-templates.md#first-timer-only-issues) for triage.
```
Comment (Sembauke) [inline docs/moderator-handbook.md:44]:
```suggestion
For triaging a trivial issue such as a typo fix, it is recommended to apply a "first timers only" label along with additional information. You can utilize the [reply template](reply-templates.md#first-timer-only-issues) provided for this purpose.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: docs/how-to-work-on-the-component-library.md (hunk 1 above)
Comment (jeremylt) [reply docs/how-to-work-on-the-component-library.md:97]: I think I like the `for` in there because it makes the link part of a complete sentence.

Comment (naomi-lgbt) [reply docs/how-to-work-on-the-component-library.md:97]: I'm in agreement with Jeremy - the current structure is grammatically correct.

Comment (Sembauke) [reply docs/how-to-work-on-the-component-library.md:97]: Feel free to revert it back

File: docs/security-hall-of-fame.md
Comment (Nirajn2311) [inline docs/security-hall-of-fame.md:1]: This page cannot be found in the sidebar. Is that intentional?

**Review by Nirajn2311 — CHANGES_REQUESTED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**